frontend/src/pages/CompareV2.tsx

```diff
@@ -313,10 +375,20 @@ function getDropdownItems(filteredRunArtifacts: RunArtifact[]) {
   return dropdownItems;
 }
 
-function getArtifactFromSelectedItem(
+const getNamespace = (selectedItem: SelectedItem, filteredRunArtifacts: RunArtifact[]) => {
+  const selectedRun = filteredRunArtifacts.find(
+    runArtifact => runArtifact.run.run?.name === selectedItem.itemName,
+  )?.run;
+  if (selectedRun) {
+    const jsonWorkflow = JSON.parse(selectedRun.pipeline_runtime!.workflow_manifest || '{}');
+    return jsonWorkflow.metadata?.namespace;
+  }
```

Comment:
What to return if `selectedRun` is empty?

Reply:
Currently, I return nothing which was actually intentional - the relevant `getHtmlViewerConfig` function accepts either `string` or `undefined` for the `namespace`: https://github.com/kubeflow/pipelines/blob/49c3587591fc291dc4b0f85ea7339d330ffe451c/frontend/src/components/viewers/MetricsVisualizations.tsx#L568-L571

Therefore, since it seems that `namespace` is not necessarily required to retrieve the viewer configs, I planned to return nothing if no `string` could be found. However, I'm happy to modify this depending on the necessity of a `string` for `namespace`, or if it helps to clarify the intent of the `getNamespace` function.

Reply:
Thanks for explanation! We can call out the return type in function signature in this case:

`function VisualizationPanelItem(props: VisualizationPanelItemProps): string | undefined`

Reply:
Sounds good! Per https://github.com/kubeflow/pipelines/pull/7981#issuecomment-1179228054, I've made the change on my next PR: https://github.com/kubeflow/pipelines/pull/7989/commits/a5a9689b3336bf7acc301c5f17bcafba603dcfb5.